backend/src/appointment/utils.py

```diff
@@ -60,28 +60,33 @@ def decrypt(value):
 
 
 def retrieve_user_url_data(url):
-    """URL Decodes, and retrieves username, signature, and main url from /<username>/<signature>/"""
+    """URL Decodes, and retrieves username, slug/signature, and main url from /<username>/<slug/signature?>/"""
     parsed_url = parse.urlparse(url)
     split_path = [x for x in parsed_url.path.split('/') if x]
-
-    if split_path is None or len(split_path) == 0:
+    
+    # Check for general validity of the path
+    if split_path is None or len(split_path) == 0 or len(split_path) > 3:
         return False
-    # If we have more than two entries, grab the last two
-    elif len(split_path) > 2:
-        split_path = split_path[-2:]
+
+    # Normalize short and long urls to only username and slug/signature (remove the /user/ segment)
+    # FIXME: Handle edge case: A user with username='user' might make trouble here
+    if len(split_path) > 1 and split_path[0] == 'user':
```

Comment:
Why did this need to change to account for /user/ part? Was this not working as intended? The /user/ portion was always there iirc.

Reply:
It wasn't working for schedules without a slug on my end, so I thought this was caused by my local instance not having configured a short URL and always using long URLs. I made those changes to support both application states (having a short URL and having none), but please let me know if I got sth fundamentally wrong here 😅